### PR TITLE
Ds terraform intel azure tdx attestation linux vm

### PR DESCRIPTION
**Pull request by daveshrestha-intel**
This example creates an Azure VM with TDX and Intel Trust Authority attestation configuration and validation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,5 @@ override.tf.json
 
 # Include tfplan files to ignore the plan output of command: terraform plan -out=tfplan
 # example: *tfplan*
+examples/azure-linux-tdx-vm/main.tf
```
Comment (lucasmelogithub) [inline .gitignore:33]: These should not be here. If they are Git won't track them anymore.

File: examples/azure-linux-tdx-ita-attestation-vm/README.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+Once the TDX Intel Confidential Computing VM with Intel Trust Authority is created, it will validate successfull configuation of Intel Trust Authority with your ITA Token with the "ITA-Verify.tok" file. You can view the ITA_Verify.tok in the trustauthority-client/tdx-cli folder to validate.
+
+NOTE: This exampel will take about 10min to complete as there is a 5min pause while runing the ansible playbook to allow Intel Trust Authority setup to complete before completing validattion. 
+
+After the buildout is complete, you will need to assiciate your Public IP to the newely created TDX Azure isntance and configure you Network Security Group for SSH etc as needed.
+
+## Usage
+
+** See examples folder for code ./examples/azure-linux-multi-disks/main.tf **
```
Comment (lucasmelogithub) [inline examples/azure-linux-tdx-ita-attestation-vm/README.md:42]: Update

File: examples/azure-linux-tdx-ita-attestation-vm/README.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+```hcl
+
+################################################################################################
+# Replace the default values with your own values - your trustauthority_api_key (ITA token)    #
+################################################################################################
+# Variable for trustauthority_api_key
+variable "trustauthority_api_key" {
+  description = "trustauthority_api_key"
+    type        = string
+    default     = "<YOUR ITA TOKEN HERE>="
```
Comment (lucasmelogithub) [inline examples/azure-linux-tdx-ita-attestation-vm/README.md:55]: Comment out this line and make bring up more attention to the need to modify it.

File: examples/azure-linux-tdx-ita-attestation-vm/cloud_init.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  - git clone https://github.com/intel/optimized-cloud-recipes.git /tmp/optimized-cloud-recipes
+  - cd /tmp/optimized-cloud-recipes/recipes/intel-azure-linux-tdx-ita 
+  - ansible-playbook recipe.yml
+  - echo 'First command set executed successfully!'
+  - sleep 300  # Pause for 5 minutes (300 seconds)
+  - cd /trustauthority-client/tdx-cli
+  - sudo make cli
+  - sudo cp /home/adminuser/intel-ita/config.json /trustauthority-client/tdx-cli/config.json
+  - sudo ./trustauthority-cli token --config config.json --no-eventlog > ITA-Token.tok
+  - sudo ./trustauthority-cli verify --config config.json --token $(cat ITA-Token.tok) > ITA-Verify.tok
```
Comment (lucasmelogithub) [inline examples/azure-linux-tdx-ita-attestation-vm/cloud_init.yml:45]: Is sudo required ? I don't believe it is.

File: examples/azure-linux-tdx-ita-attestation-vm/variables.tf [hunk 5]
```diff
@@ -0,0 +1,64 @@
+############################################################################################
+# Replace the default values with your own values - your trustauthority_api_key (token)    #
+############################################################################################
+# Variable for trustauthority_api_key
+variable "trustauthority_api_key" {
+  description = "trustauthority_api_key"
+    type        = string
+    default     = "<YOUR ITA TOKEN HERE>"
```
Comment (lucasmelogithub) [inline examples/azure-linux-tdx-ita-attestation-vm/variables.tf:8]: comment out to ensure user has to pass a value

File: .gitignore (hunk 1 above)
Comment (daveshrestha-intel) [reply .gitignore:33]: Yeah, I noticed that. I didn't put that there- I removed it now.

File: examples/azure-linux-tdx-ita-attestation-vm/README.md (hunk 2 above)
Comment (daveshrestha-intel) [reply examples/azure-linux-tdx-ita-attestation-vm/README.md:42]: done

File: examples/azure-linux-tdx-ita-attestation-vm/README.md (hunk 3 above)
Comment (daveshrestha-intel) [reply examples/azure-linux-tdx-ita-attestation-vm/README.md:55]: Done

File: examples/azure-linux-tdx-ita-attestation-vm/variables.tf (hunk 5 above)
Comment (daveshrestha-intel) [reply examples/azure-linux-tdx-ita-attestation-vm/variables.tf:8]: Done

File: examples/azure-linux-tdx-ita-attestation-vm/cloud_init.yml (hunk 4 above)
Comment (daveshrestha-intel) [reply examples/azure-linux-tdx-ita-attestation-vm/cloud_init.yml:45]: the make cli command requires elivated privileges which is why i put it sudo. I tested without sudo and it fails.

**Review by lucasmelogithub — APPROVED** [review]
LGTM

**Pull request merged**